sklearn/linear_model/_base.py

```diff
@@ -673,7 +673,11 @@ def rmatvec(b):
                 )
                 self.coef_ = np.vstack([out[0] for out in outs])
         else:
-            self.coef_, _, self.rank_, self.singular_ = linalg.lstsq(X, y)
```

Comment:
In case that we keep scipy `lstsq`, it might be safe here to pass `check_finite=False` since we validate `X` and `y` or do we consider that the centering could trigger `inf` and `nan` for some reason?

Reply:
Seems like a good idea to improve the performance, I'll add the option in the benchmark. After validating `X,y` and checking `sample_weight` I think we are safe ? Except that `_check_sample_weight(ensure_non_negative=True)` still allows all zero weights, so we should probably raise an error in that case.